src/java.base/share/classes/jdk/internal/foreign/abi/BindingSpecializer.java

```diff
@@ -682,8 +682,21 @@ private void emitCast(Binding.Cast cast) {
             popType(int.class);
 
             if (toType == boolean.class) {
-                emitConst(0b1);
+                // implement least significant byte non-zero test
+                Label falseLabel = new Label();
```

Comment:
Optional: couldn't we call `Utils.byteToBoolean`?

Reply:
Yes we can, good point! (stuck in my head with that we couldn't call internal code)